onadata/apps/api/viewsets/xform_viewset.py

```diff
@@ -708,6 +710,11 @@ def partial_update(self, request, *args, **kwargs):
         if request.FILES or set(['xls_url',
                                  'dropbox_xls_url',
                                  'text_xls_form']) & set(request.data):
+            # send form update notification
+            send_message(
+                instance_id=self.object.id, target_id=self.object.id,
+                target_type=XFORM, user=self.request.user,
+                message_verb=FORM_UPDATED)
```

Comment:
Is there a chance that the `_try_update_xlsform` might fail?

Reply:
_Mainly concerned with sending the notification before confirming that the XForm has been updated..._

Reply:
Makes sense. Made changes